src/Angor/Shared/WalletOperations.cs

```diff
@@ -35,7 +36,84 @@ public string GenerateWalletWords()
+    {
+        Network network = _networkConfiguration.GetNetwork();
+        var nbitcoinNetwork = NetworkMapper.Map(network);
+
+        changeAddress = changeAddress ?? accountInfo.GetNextChangeReceiveAddress();
+
+        var utxoDataWithPaths = FindOutputsForTransaction((long)transaction.Outputs.Sum(_ => _.Value), accountInfo);
+        var coins = utxoDataWithPaths.Select(u => new Coin(uint256.Parse(u.UtxoData.outpoint.transactionId), (uint)u.UtxoData.outpoint.outputIndex, Money.Satoshis(u.UtxoData.value), Script.FromHex(u.UtxoData.scriptHex))).ToList();
+
+        if (!coins.Any())
+            throw new ApplicationException("No coins found to fund the transaction.");
+
+        var builder = new TransactionBuilder(network)
+            .AddCoins(coins)
+            .SetChange(BitcoinAddress.Create(changeAddress, network))
+            .ContinueToBuild(transaction) // Add the predefined outputs
+            .SendEstimatedFees(new FeeRate(Money.Satoshis(feeRate)))
+            .CoverTheRest(); // Ensure enough input value covers outputs + fee, adjusting change if needed
+
+        var unsignedTx = builder.BuildTransaction(false);
+
+        var psbt = NBitcoin.PSBT.FromTransaction(NBitcoin.Transaction.Parse(unsignedTx.ToHex(), nbitcoinNetwork), nbitcoinNetwork);
+
+        NBitcoin.ExtPubKey accountExtPubKey = NBitcoin.ExtPubKey.Parse(accountInfo.RootExtPubKey, nbitcoinNetwork);
+
+        for (int i = 0; i < unsignedTx.Inputs.Count; i++)
+        {
+            var input = unsignedTx.Inputs[i];
+            var utxoInfo = utxoDataWithPaths.FirstOrDefault(u => u.UtxoData.outpoint.ToString() == input.PrevOut.ToString());
+
+            if (utxoInfo == null)
+                throw new InvalidOperationException($"Could not find UTXO information for input {input.PrevOut}");
+
+            psbt.Inputs[i].WitnessUtxo = new NBitcoin.TxOut(NBitcoin.Money.Satoshis(utxoInfo.UtxoData.value), NBitcoin.Script.FromHex(utxoInfo.UtxoData.scriptHex));
+
+            var keyPath = new NBitcoin.KeyPath(utxoInfo.HdPath);
+            var rootedKeyPath = new NBitcoin.RootedKeyPath(accountExtPubKey, keyPath);
+
+            var pubKey = _hdOperations.GeneratePublicKey(ExtPubKey.Parse(accountInfo.ExtPubKey, network), (int)keyPath.Indexes[4], keyPath.Indexes[3] == 1);
+            var path = _hdOperations.CreateHdPath(Purpose, network.Consensus.CoinType, AccountIndex, keyPath.Indexes[3] == 1, (int)keyPath.Indexes[4]);
+
+            if (path != utxoInfo.HdPath)
+                throw new InvalidOperationException($"Path does not match {path} {utxoInfo.HdPath}");
+
+            psbt.Inputs[i].HDKeyPaths.Add(new NBitcoin.PubKey(pubKey.ToBytes()), rootedKeyPath);
+        }
+
+        return new PsbtWrapper { PsbtHex = psbt.ToHex() };
+    }
+
+    public TransactionInfo SignPsbt(PsbtWrapper psbtWrapper, WalletWords walletWords)
```

Comment:
Will we be sending the wallet words to the other wallets as well?

Reply:
no but assuming this code will be called form the desktop or a signing device then the words will be available there

Reply:
Perhaps we should have a different approach and inject the access to the Wallet words in our implementation and otherwise use external wallet with the same Interface

Reply:
I dont know yet, it will be more clear when we come to implement HW